### PR TITLE
Active window size and position

### PR DESCRIPTION
**Pull request by sam-dixon**
This implements a new driver API function to get the size and position of the active window. Tested on macOS and Windows, but not yet implemented for Linux

### CODE REVIEW

File: src/mac.cpp [hunk 1]
```diff
@@ -342,9 +342,31 @@ bool ActiveApplicationIsSandboxed() {
...
+        std::get<0>(result) = [[bounds objectForKey:@"X"] intValue];
+        std::get<1>(result) = [[bounds objectForKey:@"Y"] intValue];
+        std::get<2>(result) = [[bounds objectForKey:@"Height"] intValue];
+        std::get<3>(result) = [[bounds objectForKey:@"Width"] intValue];
+        CFRelease(windows);
+        return result;
+      }
+    }
+  }
+  return result;
```
Comment (tmacwill) [inline src/mac.cpp:364]: I think we need a `CFRelease` here, to make sure we don't have a memory leak if no window is found.

File: src/mac.cpp [hunk 2]
```diff
@@ -342,9 +342,31 @@ bool ActiveApplicationIsSandboxed() {
   return isSandboxed;
 }
 
+std::tuple<int, int, int, int> GetActiveApplicationWindowBounds() {
+  NSArray* windows = (NSArray*)CGWindowListCopyWindowInfo(
+    kCGWindowListOptionOnScreenOnly | kCGWindowListExcludeDesktopElements, kCGNullWindowID);
```
Comment (tmacwill) [inline src/mac.cpp:347]: should this also be `kCGWindowListOptionAll` like the other instances of this call? if so, it's probably worth creating a small utility function here, so we have a consistent way of getting the windows to iterate over.

**Review by tmacwill — CHANGES_REQUESTED** [review]
thanks for doing this! few small things inline

File: src/mac.cpp (hunk 2 above)
Comment (sam-dixon) [reply src/mac.cpp]: It should, but with an extra check to make sure the window we're getting the information for is visible and the main window. I added this and the utility function in 00c4442

**Review by tmacwill — APPROVED** [review]
looks great, thanks! let's make sure to add Linux support before we announce (unless it's not possible there), to make sure the API is consistent across platforms.

**Comment by sam-dixon** [comment]
Added Linux compatibility -- sort of. In testing, I found that not all windows are detected and sometimes the position is incorrect, but I'm not sure how much more I can do here.

**Pull request merged**